PAPP-35185: Network, policy, rule and device related actions for cloud and on-prem (#1)
splunk-soar-connectors/ciscosecurefirewall: README.md

# Cisco Secure Firewall

Publisher: Splunk \
Connector Version: 1.0.0 \
Product Vendor: Cisco Systems \
Product Name: Cisco Secure Firewall \
Minimum Product Version: 6.3.0

> This app interfaces with Cisco Firepower devices to adds, updates and deletes network objects, network object groups, access policies and access rules

Reviewer: ```suggestion
This app interfaces with Cisco Firepower devices to add, update and delete network objects, network object groups, access policies and access rules
```

This connector supports both cloud and on-prem delivered FMC. Below are the steps for connecting to both

## Connecting to a cloud delivered FMC

1. On Cisco Security Cloud Control navigate to User Management
1. Create a new Api Only User with an Admin role
1. Copy the Api key and enter it in the "Api key for cloud delivered FMC" input box in the SOAR Asset Settings page
1. Specfiy Cloud for the type of FMC you are connecting to
1. Specify your region in the "Region your Cisco Security Cloud Control is deployed in" input box and click Save

## Connecting to an on-prem delivered FMC

1. On the SOAR asset setting page select On-prem for the type of FMC you are connecting to

> 1. Specify the device ip/hostname of your on-prem FMC along with the username and password used ot login to FMC

Reviewer: ```suggestion
1. Specify the device ip/hostname of your on-prem FMC along with the username and password used to login to FMC
```

> **Note** that you can optionally specify a default firepower domain that will be queried. You an overide this domain when running an action. In addition, cloud versions of FMC only support the default domain, to achieve multi tenancy you must use seperate tenants.

Reviewer: ```suggestion
**Note** that you can optionally specify a default firepower domain that will be queried. You can override this domain when running an action. In addition, cloud versions of FMC only support the default domain. To achieve multi tenancy you must use separate tenants.
```

[list network objects](#action-list-network-objects) - List network object in FMC \
[create network object](#action-create-network-object) - Creates a network object in FMC \
[update network object](#action-update-network-object) - Updates a network object in FMC \
[delete network object](#action-delete-network-object) - Deletes a network object in FMC \
[get network group objects](#action-get-network-group-objects) - Gets all network group objects in FMC host or a specfic network group \
[create network group object](#action-create-network-group-object) - Create a network group object \
[update network group object](#action-update-network-group-object) - Update a network group object \
[delete network group object](#action-delete-network-group-object) - Delete a network group object \
[get access control policies](#action-get-access-control-policies) - Gets all or a particular access control policy in the FMC host for a particular domain \
[create access control policy](#action-create-access-control-policy) - Create an access control policy \
[update access control policy](#action-update-access-control-policy) - Update an access control policy \

> [delete access control policies](#action-delete-access-control-policies) - Deleted the specified access control policy \

Reviewer: ```suggestion
[delete access control policies](#action-delete-access-control-policies) - Deletes the specified access control policy \
```

[get access control rules](#action-get-access-control-rules) - Gets all access control rules associated with a particular access control policy \
[create access control rule](#action-create-access-control-rule) - Creates an access control rule associated with a particular access control policy \
[update access control rule](#action-update-access-control-rule) - Updates an access control rule associated with a particular access control policy \
[delete access control rules](#action-delete-access-control-rules) - Deletes access control rule associated with a particular access control policy \
[list intrusion policies](#action-list-intrusion-policies) - Gets all intrusion polcies in the FMC host for a particular domain \
[create intrusion policy](#action-create-intrusion-policy) - Create an intrusion policy \
[update intrusion policy](#action-update-intrusion-policy) - Update an intrusion policy \

> [delete intrusion policy](#action-delete-intrusion-policy) - Deleted the specified access intrusion policy \

Reviewer: ```suggestion
[delete intrusion policy](#action-delete-intrusion-policy) - Deletes the specified access intrusion policy \
```

**domain_name** | optional | Firepower Domain. If none is specified the default domain will be queried | string | |

#### Action Output

DATA PATH | TYPE | CONTAINS | EXAMPLE VALUES
--------- | ---- | -------- | --------------
action_result.status | string | | success failed |
action_result.message | string | | |
summary.total_objects | numeric | | |
summary.total_objects_successful | numeric | | |
action_result.parameter.name | string | | |
action_result.parameter.type | string | | Network |
action_result.parameter.domain_name | string | | |
action_result.data.\*.id | string | | |
action_result.data.\*.name | string | | |
action_result.data.\*.type | string | | Network |
action_result.data.\*.links.self | string | | |
action_result.data.\*.links.parent | string | | |

## action: 'create network object'

Creates a network object in FMC

Type: **generic** \
Read only: **False**

#### Action Parameters

PARAMETER | REQUIRED | DESCRIPTION | TYPE | CONTAINS
--------- | -------- | ----------- | ---- | --------
**name** | required | Network object name | string | |
**type** | required | Network object type | string | |

> **value** | required | Value of the network object. If type if Range specify value in the following format: ip1-ip2 | string | |

Reviewer: ```suggestion
**value** | required | Value of the network object. If type is Range specify value in the following format: ip1-ip2 | string | |
```

action_result.parameter.domain_name | string | | |
action_result.data.\*.id | string | | |
action_result.data.\*.name | string | | |
action_result.data.\*.type | string | | Network |
action_result.data.\*.links.self | string | | |
action_result.data.\*.links.parent | string | | |
action_result.data.\*.value | string | | |
action_result.data.\*.metadata.domain.id | string | | |
action_result.data.\*.metadata.domain.name | string | | |
action_result.data.\*.metadata.domain.type | string | | |
action_result.data.\*.metadata.ipType | string | | |
action_result.data.\*.metadata.domain.lastUser.name | string | | |
action_result.data.\*.metadata.domain.timestamp | numeric | | |
action_result.data.\*.metadata.domain.parentType | string | | |
action_result.data.\*.metadata.overridable | boolean | | |

## action: 'update network object'

Updates a network object in FMC

Type: **generic** \
Read only: **False**

#### Action Parameters

PARAMETER | REQUIRED | DESCRIPTION | TYPE | CONTAINS
--------- | -------- | ----------- | ---- | --------
**object_id** | required | Network object id | string | |
**name** | optional | Network object name | string | |

> **type** | optional | Network object type. Note this cannot change and is only used to identify the network object value you'd liek to update. | string | |

Reviewer: ```suggestion
**type** | optional | Network object type. Note this cannot change and is only used to identify the network object value you'd like to update. | string | |
```

action_result.data.\*.type | string | | NetworkGroup |
action_result.data.\*.links.self | string | | |
action_result.data.\*.objects.id | string | | |
action_result.data.\*.objects.name | string | | |
action_result.data.\*.objects.type | string | | Network |
action_result.data.\*.metadata.domain.id | string | | |
action_result.data.\*.metadata.domain.name | string | | |
action_result.data.\*.metadata.domain.type | string | | |
action_result.data.\*.metadata.domain.lastUser.name | string | | |
action_result.data.\*.metadata.domain.timestamp | numeric | | |
action_result.data.\*.metadata.domain.parentType | string | | |
action_result.data.\*.metadata.overridable | boolean | | |

## action: 'update network group object'

Update a network group object

Type: **generic** \
Read only: **False**

#### Action Parameters

PARAMETER | REQUIRED | DESCRIPTION | TYPE | CONTAINS
--------- | -------- | ----------- | ---- | --------
**network_group_id** | required | Network group to update | string | |
**name** | optional | Name of the network group | string | |
**network_object_ids_to_add** | optional | Network objects to add to the group. Note these ids must already exist in FMC | string | |

> **network_object_ids_to_remove** | optional | Network objects to remove frin the group. | string | |

Reviewer: ```suggestion
**network_object_ids_to_remove** | optional | Network objects to remove from the group. | string | |
```
